tests/cmdline/commands/test_daemon.py

```diff
@@ -86,7 +86,7 @@ def test_daemon_restart(self):
         finally:
             self.daemon_client.stop_daemon(wait=True)
 
-    @pytest.mark.skip(reason='Test fails non-deterministically; see issue #3051.')
+    @pytest.mark.flaky(reruns=2)
```

Comment:
Maybe we should keep the reference to the issue number in a comment?

Reply:
I agree - I've now added this info to all flaky tests